File: solutions/4162-minimum-number-of-operations-to-have-distinct-elements/solution.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        map<int,int>m;
        for(int i=nums.size()-1;i>=0;i--)
        {
            if(m.contains(nums[i]))
                return (i+3)/3;
            m[nums[i]]=1;
        }
        return 0;
    }
};

File: solutions/4162-minimum-number-of-operations-to-have-distinct-elements/solution.java
class Solution {
    public int minOperations(int[] nums) {
        Map<Integer,Integer>map=new HashMap<>();
        for(int i=nums.length-1;i>=0;i--)
        {
            if(map.containsKey(nums[i]))
                return (i+3)/3;
            map.put(nums[i],1);
        }
        return 0;
    }
}
